lib/active_merchant/billing/gateways/versa_pay.rb

```diff
@@ -154,35 +171,40 @@ def parse(body)
         }.with_indifferent_access
       end
 
-      def commit(action, post)
-        raw_response = ssl_post(url(action), post.to_json, request_headers)
+      def commit(action, post, method = :post, options = {})
+        raw_response = ssl_request(method, url(action, method, options), post.to_json, request_headers)
         response = parse(raw_response)
         first_transaction = response['transactions']&.first
 
         Response.new(
-          success_from(response),
-          message_from(response),
+          success_from(response, action),
+          message_from(response, action),
           response,
           authorization: authorization_from(response),
           avs_result: AVSResult.new(code: dig_avs_code(first_transaction)),
           cvv_result: CVVResult.new(dig_cvv_code(first_transaction)),
           test: test?,
-          error_code: error_code_from(response)
+          error_code: error_code_from(response, action)
         )
       end
 
-      def success_from(response)
-        response['success'] || false
+      def success_from(response, action)
+        case action
+        when 'wallets'
+          response['wallet_token'] || response['fund_token'] || false
+        else
+          response['success'] || false
+        end
       end
 
-      def message_from(response)
-        return 'Succeeded' if success_from(response)
+      def message_from(response, action)
+        return 'Succeeded' if success_from(response, action)
 
         first_transaction = response['transactions']&.first
         gateway_response_errors = gateway_errors_message(response)
 
         response_message = {
-          errors: response['errors']&.join(', ').presence,
+          errors: response.dig('errors')&.join(', ').presence,
```

Comment:
💬 Comment:

Whats the point of `dig` vs `[ ]` ?

Reply:
It was because of a different kind of error, but it was solved differently, it can be removed.